### PR TITLE
TDL-20061: Added missing fields

### PR DESCRIPTION
**Pull request by savan-chovatiya**
# Description of change
TDL-20061: Update/add missing fields
- Added fields are which are missing in the schema but present in the API doc.
- Updated datatype for some of the fields as per API doc.
- Added reference to some fields to dry up the schema

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

**Review by harshpatel4crest — APPROVED** [review]

File: tap_quickbooks/schemas/bills.json [hunk 1]
```diff
@@ -76,9 +76,10 @@
             ]
           },
           "LineNum": {
```
Comment (dbshah1212) [inline tap_quickbooks/schemas/bills.json:78]: Do we need a format change for the Line Number?

**Review by dbshah1212 — CHANGES_REQUESTED** [review]
Why do we need to change the LineNum format from integer to decimal? Other changes look good.

File: tap_quickbooks/schemas/bills.json (hunk 1 above)
Comment (savan-chovatiya) [reply tap_quickbooks/schemas/bills.json:78]: The API doc suggests that the data type is Decimal. However, it also mentioned that it is a positive integer. https://developer.intuit.com/app/developer/qbo/docs/api/accounting/most-commonly-used/bill

File: tap_quickbooks/schemas/shared/group_line.json [hunk 2]
```diff
@@ -45,7 +45,8 @@
                 }
               },
               "Qty": {
-                "type": ["null", "integer"]
+                "format": "singer.decimal",
```
Comment (dbshah1212) [inline tap_quickbooks/schemas/shared/group_line.json:48]: Same of Qty if it is quantity then it won't be in decimal.

File: tap_quickbooks/schemas/time_activities.json [hunk 3]
```diff
@@ -1,9 +1,10 @@
 {
   "properties": {
     "HourlyRate": {
+      "format": "singer.decimal",
       "type": [
         "null",
-        "integer"
+        "string"
```
Comment (dbshah1212) [inline tap_quickbooks/schemas/time_activities.json:7]: Same for HourlyRate, if it is quantity then it won't be in decimal.

File: tap_quickbooks/schemas/bills.json (hunk 1 above)
Comment (savan-chovatiya) [reply tap_quickbooks/schemas/bills.json]: Reverted back changes for LineNum.

File: tap_quickbooks/schemas/shared/group_line.json (hunk 2 above)
Comment (savan-chovatiya) [reply tap_quickbooks/schemas/shared/group_line.json]: Reverted back changes for Qty.

File: tap_quickbooks/schemas/time_activities.json (hunk 3 above)
Comment (savan-chovatiya) [reply tap_quickbooks/schemas/time_activities.json]: Reverted back changes for HourlyRate.

File: tap_quickbooks/schemas/items.json [hunk 4]
```diff
@@ -77,15 +77,17 @@
       ]
     },
     "QtyOnHand": {
```
Comment (dbshah1212) [inline tap_quickbooks/schemas/items.json:79]: If it is quantity then it won't be in decimal.

Comment (savan-chovatiya) [reply tap_quickbooks/schemas/items.json]: Reverted back QtyOnHand to an integer.

**Review by dbshah1212 — APPROVED** [review]

**Review by shantanu73 — APPROVED** [review]

**Review by rdeshmukh15 — APPROVED** [review]

**Pull request merged**